remove packedDependencies and custom package script

jackkav opened the pull request:
uses electron-builder conventions to avoid maintaining the packedDependencies list and release.ts
the subtle conventional change is that insomnia-app/package.json devDependencies are all bundled by vite and dependencies will be installed inside the final package as before.
<!--
Please open an [Issue](https://github.com/kong/insomnia/issues/new) first to discuss new
features or non-trivial changes. Please provide as much detail as possible on the change as
possible including general description, implementation details, potential shortcomings, etc.

If this PR closes an issue, please mention "Closes #XX" where #XX is the issue number.

If this PR fixes a bug or regression, please make sure to add a test.
-->

jackkav committed:
use electron-builder

jackkav committed:
hard links workaround

dimitropoulos commented on packages/insomnia-app/.electron-builder.config.js:
```diff
@@ -0,0 +1,92 @@
+/**
+ * @type {import('electron-builder').Configuration}
```
good call!  nice to have validation!

dimitropoulos commented on packages/insomnia-app/.electron-builder.config.js:
```diff
@@ -0,0 +1,92 @@
+// NOTE: USE_HARD_LINKS
+// https://github.com/electron-userland/electron-builder/issues/4594#issuecomment-574653870
+const config = {
+    appId: "com.insomnia.app",
```
this was `__APP_ID__` before.  is there any significance to it being `com.insomnia.app` now?

jackkav replied:
thats just what package.ts replaced it with from config.json AFAIK
```js
const rawConfig = JSON.stringify(electronBuilderConfig, null, 2)
    .replace(/__APP_ID__/g, appId)
    .replace(/__BINARY_PREFIX__/g, binaryPrefix)
    .replace(/__EXECUTABLE_NAME__/g, executableName)
    .replace(/__GITHUB_OWNER__/g, githubOrg)
    .replace(/__GITHUB_REPO__/g, githubRepo)
    .replace(/__ICON_URL__/g, icon)
    .replace(/__SYNOPSIS__/g, synopsis);
```
the original package.ts may have some behaviors that I didn't capture in the electron-builder.config.js
like mac universal

dimitropoulos replied:
riggghhhhttt righhhhtttt forgot when I wrote this and un-forgot later but forgot to delete my comment -> thanks for confirming!

dimitropoulos commented on packages/insomnia-app/config/electronbuilder.json:
```diff
@@ -1,89 +0,0 @@
-  },
-  "linux": {
-    "artifactName": "__BINARY_PREFIX__-${version}.${ext}",
-    "executableName": "__EXECUTABLE_NAME__",
```
should we also then delete this from `config.json`?

jackkav replied:
I'm not sure if config.json what the distinction is between it and constants.ts, perhaps just moving anything used into constants.ts would be a more convenient grouping.

dimitropoulos committed:
extracts binary prefix back to variable (but this time, local)

to avoid the future footgun of one of them drifting by accident

dimitropoulos committed:
enables linting for config file

which, coincidentally, seems to require not using the `.` prefix (https://github.com/microsoft/TypeScript/issues/13399#issuecomment-271939777), and as far as I can tell it's not needed and it builds fine on my machine (at least) without it.

dimitropoulos committed:
deletes (now unused) config.json keys

dimitropoulos reviewed (commented):
What a time to be alive! This is exciting!

I spent a good deal of time just now looking at this and trying to find a hole in it, but I see none.  Now, I'm not an expert on electron-builder, but this sure does seem to be a more "normal" way to use it.

I added a few (very very minor) tweaks (see commit message and description for more on what or why).

_____

The only thing that is of some concern is that I noticed that the before-and-after of the output is different, somewhat significantly so.  Can someone else take a look at this?

Before this PR I get something like this:
```
insomnia/packages/insomnia-app/dist$ find . -not -path "*node_modules*" -not -path "*locales*"
.
./Insomnia.Core-2022.3.0-beta.3.AppImage
./Insomnia.Core-2022.3.0-beta.3.snap
./Insomnia.Core-2022.3.0-beta.3.tar.gz
./.icon-set
./.icon-set/icon_48x48.png
./.icon-set/icon_256.png
./.icon-set/icon_128x128.png
./.icon-set/icon_512.png
./.icon-set/icon_32x32.png
./.icon-set/icon_64x64.png
./.icon-set/icon_16x16.png
./builder-effective-config.yaml
./linux-unpacked
./linux-unpacked/LICENSE.electron.txt
./linux-unpacked/chrome-sandbox
./linux-unpacked/v8_context_snapshot.bin
./linux-unpacked/libvulkan.so.1
./linux-unpacked/icudtl.dat
./linux-unpacked/snapshot_blob.bin
./linux-unpacked/chrome_200_percent.pak
./linux-unpacked/libGLESv2.so
./linux-unpacked/resources
./linux-unpacked/resources/bin
./linux-unpacked/resources/bin/yarn-standalone.js
./linux-unpacked/resources/opensource-licenses.txt
./linux-unpacked/resources/app.asar
./linux-unpacked/swiftshader
./linux-unpacked/swiftshader/libGLESv2.so
./linux-unpacked/swiftshader/libEGL.so
./linux-unpacked/libffmpeg.so
./linux-unpacked/resources.pak
./linux-unpacked/chrome_crashpad_handler
./linux-unpacked/libvk_swiftshader.so
./linux-unpacked/insomnia
./linux-unpacked/chrome_100_percent.pak
./linux-unpacked/libEGL.so
./linux-unpacked/vk_swiftshader_icd.json
./linux-unpacked/LICENSES.chromium.html

```

But after this PR I get something like this:
```
insomnia/packages/insomnia-app/dist$ find . -not -path "*node_modules*" -not -path "*locales*"
.
./builder-debug.yml
./builder-effective-config.yaml
./linux-unpacked
./linux-unpacked/LICENSE.electron.txt
./linux-unpacked/chrome-sandbox
./linux-unpacked/v8_context_snapshot.bin
./linux-unpacked/libvulkan.so.1
./linux-unpacked/icudtl.dat
./linux-unpacked/snapshot_blob.bin
./linux-unpacked/chrome_200_percent.pak
./linux-unpacked/libGLESv2.so
./linux-unpacked/resources
./linux-unpacked/resources/bin
./linux-unpacked/resources/bin/yarn-standalone.js
./linux-unpacked/resources/app
./linux-unpacked/resources/app/app
./linux-unpacked/resources/app/app/chart.383d9cac.svg
./linux-unpacked/resources/app/app/install-spinner.gif
./linux-unpacked/resources/app/app/index.c47a3bbf.js
./linux-unpacked/resources/app/app/fontawesome-webfont.7dacf83f.woff2
./linux-unpacked/resources/app/app/opensource-licenses.txt
./linux-unpacked/resources/app/app/static
./linux-unpacked/resources/app/app/static/insomnia-core-logo_16x.png
./linux-unpacked/resources/app/app/static/entitlements.mac.inherit.plist
./linux-unpacked/resources/app/app/icon.icns
./linux-unpacked/resources/app/app/main.min.js.map
./linux-unpacked/resources/app/app/icon.ico
./linux-unpacked/resources/app/app/insomnia-logo.a19543fc.svg
./linux-unpacked/resources/app/app/preload.js
./linux-unpacked/resources/app/app/background.png
./linux-unpacked/resources/app/app/index.html
./linux-unpacked/resources/app/app/icn-eye.070cdb80.svg
./linux-unpacked/resources/app/app/main.min.js
./linux-unpacked/resources/app/app/index.c9a30e89.css
./linux-unpacked/resources/app/app/yarn-standalone.js
./linux-unpacked/resources/app/package.json
./linux-unpacked/resources/opensource-licenses.txt
./linux-unpacked/swiftshader
./linux-unpacked/swiftshader/libGLESv2.so
./linux-unpacked/swiftshader/libEGL.so
./linux-unpacked/libffmpeg.so
./linux-unpacked/resources.pak
./linux-unpacked/chrome_crashpad_handler
./linux-unpacked/libvk_swiftshader.so
./linux-unpacked/insomnia
./linux-unpacked/chrome_100_percent.pak
./linux-unpacked/libEGL.so
./linux-unpacked/vk_swiftshader_icd.json
./linux-unpacked/LICENSES.chromium.html
```

Of most concern for me is that the `.AppImage` files are gone now, which was something I used a decent amount in the past.

jackkav commented:
@dimitropoulos thanks for the thorough review.
part of the diff is explained by `-config.asar=false` which helped me determine the appropriate electron config for files and directories. if this is removed we should see an asar binary instead of the app folder, theres still some more to figure out with icons i suspect. feel free to continue if it interests you.

jackkav committed:
fix diff hopefully

jackkav committed:
fix lint

dimitropoulos commented on packages/insomnia-app/electron-builder.config.js:
```diff
@@ -87,8 +88,15 @@ const config = {
+const { env: { BUILD_TARGETS }, platform } = process
+const targets = BUILD_TARGETS?.split(',')
+if (platform && targets) {
+  console.log('overriding build targets to: ', targets)
```
awesome, thanks!

DMarby committed:
Make lerna ignore private packages for version bumps

And stop referencing the build folder in lerna

DMarby committed:
Make macOS produce universal builds and re-enable rpm for linux

DMarby committed:
Set the correct app version

DMarby committed:
Use the same package name as previously for electron-builder

DMarby committed:
Read app version from package.json

DMarby committed:
Make app-bump-version update package.json and package-lock.json

johnwchadwick reviewed (approved):
LGTM

DMarby committed:
Fix tests

DMarby committed:
Fix linter errors

DMarby committed:
Fix getTempDir

DMarby committed:
Clean dist folder

DMarby committed:
Fix app includes and build targets override

DMarby committed:
Remove explicit electron version from electron-builder

gatzjames reviewed (approved):
🚀

DMarby committed:
Update package-lock files

DMarby committed:
Explicitly set the app data folder for development as well
